src/VisualStudio/Core/Impl/RoslynVisualStudioWorkspace.cs

```diff
@@ -28,17 +30,23 @@ namespace Microsoft.VisualStudio.LanguageServices
     [Export(typeof(VisualStudioWorkspaceImpl))]
     internal class RoslynVisualStudioWorkspace : VisualStudioWorkspaceImpl
     {
+        private readonly IThreadingContext _threadingContext;
         private readonly IEnumerable<Lazy<IStreamingFindUsagesPresenter>> _streamingPresenters;
+        private readonly IEnumerable<Lazy<IExternalNavigationService>> _externalNavigationServices;
 
         [ImportingConstructor]
         private RoslynVisualStudioWorkspace(
+            IThreadingContext threadingContext,
```

Comment:
Rather than grabbing another copy here, the base class (VisualStudioWorkspaceImpl) is already grabbing IThreadingContext and stashing it into a private field. I'd propose just:

1. Switch the private field to a protected readonly property.
2. Just use that property in the goto def function.

Reply:
Done.